internal/handlers/short_test.go

```diff
@@ -0,0 +1,137 @@
+package handlers
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+
+	"github.com/go-chi/chi/v5"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Parse short URLs", func() {
```

Comment:
I'd prefer to use `Describe` to identify the function under test and split different groups into different `Describe` or `Context` blocks.

So this would be `Describe("idFromJWT" ...` containing tests like:
`succeeds with a JWT with an infra env ID`
`fails with an invalid JWT`
`fails without a valid claim`
etc.

The test name along with the containing contexts and describe blocks should tell you the function, scenario, and expected outcome in something approximating English.